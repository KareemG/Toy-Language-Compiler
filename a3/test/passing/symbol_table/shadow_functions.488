{
    function test : integer
    {
        return with 5
    }

    function test2 : integer
    {
        function test(a : integer, b : boolean) : integer
        {
            if b then
                return with a
            else
                return with 10
        }

        test
    }

    var x : integer

    x := test2
}