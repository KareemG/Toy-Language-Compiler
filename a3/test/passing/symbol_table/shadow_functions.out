{
 Symbols:
 x: integer
 test2: integer
 test: integer
 test : integer{
 Symbols:
 }
 test2 : integer{
 Symbols:
 test: integer
 test : integer ( a: integer, b: boolean ) {
 Symbols:
 }
 }